token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,177 @@
+"""
+A module discovers and invoke Token Dispenser Service (TDS) lambda
+"""
+import logging
+import json
+import re
+from typing import List
+from botocore.exceptions import ClientError
+import boto3
+lambda_client = boto3.client('lambda')
+ssm = boto3.client('ssm')
+# ssm parameter store default path for TDS ARN
+_DEFAULT_SSM_PATH_: str = '/service/token-dispenser'
+logger = logging.getLogger(__name__)
+
+
+def get_parameter_by_name(name: str) -> str:
+    """
+
+    Args:
+        name: the ssm name point to TDS arn
+
+    Returns: tds arn as string
+    """
+    try:
+        if not name.endswith('/'):  # Check if it's a specific parameter name
+            response = ssm.get_parameter(
+                Name=name
+            )
+        return response['Parameter']['Value']
+    except KeyError as key_error:
+        logger.exception("Missing Parameter or Value key with name:%s response:%s",
+                         name, response)
+        raise key_error
```

Comment:
Can we explicitly check if the response contains no parameters and then raise an exception that says no parameter was found instead of catching that KeyError?